cunumeric/module.py

```diff
@@ -7545,6 +7545,144 @@ def var(
+    y : optional ndarray
+        Additional set of observations to be appended to m
+    rowvar : bool
+        If true, each row is considered a random variable
+    bias :
+        Default False represents normalization by (N - 1).
+        If, true set normalization to N
+    ddof : int, optional
+        “Delta Degrees of Freedom”: the divisor used in the calculation is
+        N - ddof, where N represents the number of elements. By default
+        ddof is zero. If set, overwrites divisor normalization set by bias.
+    fweights : optional ndarray
+        A weights matrix applied to each random variable to specify
+        repeated observations over a given sample.
+    aweights : optional ndarray
+        A weights matrix applied to each random variable to specify
+        relative importance of each observation.
+
+    Returns
+    -------
+    a : ndarray, see dtype parameter above
+        If `out=None`, returns a new array of the same dtype as above
+        containing the variance values, otherwise a reference to the output
+        array is returned.
+
+    See Also
+    --------
+    numpy.cov
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+    """
+    # Check inputs
+    if ddof is not None and ddof != int(ddof):
+        raise ValueError("ddof must be integer")
+
+    # Handles complex arrays too
+    if m.ndim > 2:
+        raise ValueError("m has more than 2 dimensions")
+
+    if y is not None and y.ndim > 2:
+        raise ValueError("y has more than 2 dimensions")
+
+    if dtype is None:
+        dtype = dtype = np.dtype(np.float64)
+    X = array(m, ndmin=2, dtype=dtype)
+    if not rowvar and X.shape[0] != 1:
+        X = X.T
+    if X.shape[0] == 0:
+        return array([]).reshape(0, 0)
+    if y is not None:
+        y = array(y, copy=False, ndmin=2, dtype=dtype)
+        if not rowvar and y.shape[0] != 1:
+            y = y.T
+        X = concatenate((X, y), axis=0)
+
+    if ddof is None:
+        if bias == 0:
+            ddof = 1
+        else:
+            ddof = 0
+
+    # Get the product of frequencies and weights
+    w: Union[ndarray, None] = None
+    if fweights is not None:
+        if fweights.ndim > 1:
+            raise RuntimeError("cannot handle multidimensional fweights")
+        if fweights.shape[0] != X.shape[1]:
+            raise RuntimeError("incompatible numbers of samples and fweights")
+        if any(fweights < 0):
+            raise ValueError("fweights cannot be negative")
+        w = fweights
+    if aweights is not None:
+        if aweights.ndim > 1:
+            raise RuntimeError("cannot handle multidimensional aweights")
+        if aweights.shape[0] != X.shape[1]:
+            raise RuntimeError("incompatible numbers of samples and aweights")
+        if any(aweights < 0):
+            raise ValueError("aweights cannot be negative")
+        if w is None:
+            w = aweights
+        else:
+            w = w * aweights
```

Comment:
A problem I had with this and the /= operation mentioned lower in that it seems like the runtime can't handle these in-place operations when aweights and w are of different types, which seems to be allowed behavior in numpy. Should we instead try to do this in-place operation whenever dtype's match?

`       def _maybe_create_result(
           self,
           out: Union[ndarray, None],
           out_shape: NdShape,
           res_dtype: np.dtype[Any],
           casting: CastingKind,
           inputs: tuple[ndarray, ...],
       ) -> ndarray:
           if out is None:
               return ndarray(shape=out_shape, dtype=res_dtype, inputs=inputs)
           elif out.dtype != res_dtype:
               if not np.can_cast(res_dtype, out.dtype, casting=casting):
   >               raise TypeError(
                       f"Cannot cast ufunc '{self._name}' output from "
                       f"{res_dtype} to {out.dtype} with casting rule "
                       f"'{casting}'"
                   )
   E               TypeError: Cannot cast ufunc 'multiply' output from float64 to int64 with casting rule 'same_kind'`
   Is there something  I could do here?

Reply:
Ah I see, that makes sense. cuNumeric can't in-place multiply an int-typed array by a float, because that would need to downcast the result to make it fit into the base array. Could you instead leave a comment about this (in this instance and the `/=` below)? Just so future readers don't try to "fix" this.

Reply:
I will add that. Thank you.